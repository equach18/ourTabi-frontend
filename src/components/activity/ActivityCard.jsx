import { Link } from "react-router-dom";
import ActivityVoteButtons from "./ActivityVoteButtons";

/** Displays activity details and allows voting/editing options if the user is a member of the trip */
function ActivityCard({ activity, isTripMember }) {
  return (
    <div className="border border-zinc-300 p-4 rounded-lg shadow-sm hover:shadow-md transition bg-white">
      <div className="flex justify-between items-start gap-4">
        {/* Activity info */}
        <div className="flex-1 space-y-1">
          <div className="flex items-center gap-2">
            <h3 className="text-xl font-bold text-zinc-800">{activity.name}</h3>
            {isTripMember && (
              <Link
                to={`/trips/${activity.tripId}/activities/${activity.id}/edit`}
                className="text-yellow-500 hover:text-yellow-600 transition"
                title="Edit Activity"
              >
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  fill="none"
                  viewBox="0 0 24 24"
                  strokeWidth="1.5"
                  stroke="currentColor"
                  className="w-5 h-5"
                >
                  <path
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    d="m16.862 4.487 1.687-1.688a1.875 1.875 0 1 1 2.652 2.652L6.832 19.82a4.5 4.5 0 0 1-1.897 1.13l-2.685.8.8-2.685a4.5 4.5 0 0 1 1.13-1.897L16.863 4.487Zm0 0L19.5 7.125"
                  />
                </svg>
              </Link>
            )}
          </div>

          <p className="text-sm text-zinc-600">
            <strong>Category:</strong> {activity.category}
          </p>

          {activity.location && (
            <p className="text-sm text-zinc-600">
              <strong>Location:</strong> {activity.location}
            </p>
          )}

          {activity.description && (
            <p className="text-sm text-zinc-500">
              <strong>Description:</strong> {activity.description}
            </p>
          )}

          {activity.scheduledTime && (
            <p className="text-sm text-zinc-500">
              <strong>Scheduled:</strong>{" "}
              {new Date(activity.scheduledTime).toLocaleString(undefined, {
                dateStyle: "short",
                timeStyle: "short",
              })}
            </p>
          )}
        </div>

        {/* Voting */}
        {isTripMember && (
          <div className="flex items-center">
            <ActivityVoteButtons activityId={activity.id} />
          </div>
        )}
      </div>
    </div>
  );
}

export default ActivityCard;
