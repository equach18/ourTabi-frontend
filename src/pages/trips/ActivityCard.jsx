import { useContext } from "react";
import TripContext from "../../context/TripContext";
import { Link } from "react-router-dom";
import ActivityVoteButtons from "./ActivityVoteButtons";

function ActivityCard({ activity, isTripMember }) {
  return (
    <div className="border p-4 rounded-lg shadow-md hover:shadow-lg">
      <div className="flex justify-between items-center">
        <h3 className="text-xl font-bold">{activity.name}</h3>

        {/* Edit Button (Only for trip members) */}
        {isTripMember && (
          <Link
            to={`/trips/${activity.tripId}/activities/${activity.id}/edit`}
            className="bg-yellow-500 text-white px-3 py-1 rounded-lg"
          >
            Edit
          </Link>
        )}
      </div>

      <p className="text-gray-600">{activity.category}</p>
      <p className="text-gray-600">{activity.location}</p>
      <p className="text-gray-500">{activity.description}</p>

      {/* Activity Vote Buttons - only show if the user is a trip member */}
      {isTripMember && <ActivityVoteButtons activityId={activity.id} />}
    </div>
  );
}

export default ActivityCard;
