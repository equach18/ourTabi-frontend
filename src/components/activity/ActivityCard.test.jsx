import { screen } from "@testing-library/react";
import {
  renderWithAllProviders,
  testActivity,
} from "../../test-helpers/_testCommons";
import ActivityCard from "./ActivityCard";

describe("ActivityCard component", () => {
  it("renders ActivityCard", () => {
    renderWithAllProviders(
      <ActivityCard activity={testActivity} isTripMember={true} />
    );
    expect(screen.getByText("Kayaking")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = renderWithAllProviders(
      <ActivityCard activity={testActivity} isTripMember={true} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders location, description, and scheduled time", () => {
    renderWithAllProviders(
      <ActivityCard activity={testActivity} isTripMember={false} />
    );
    expect(screen.getByText("Location:")).toBeInTheDocument();
    expect(screen.getByText("Chicago")).toBeInTheDocument();

    expect(screen.getByText("Description:")).toBeInTheDocument();
    expect(screen.getByText("So much fun")).toBeInTheDocument();

    expect(screen.getByText(/Scheduled:/)).toBeInTheDocument();
  });

  it("does not render edit link or vote buttons if not trip member", () => {
    renderWithAllProviders(
      <ActivityCard activity={testActivity} isTripMember={false} />
    );
    expect(screen.queryByTitle("Edit Activity")).not.toBeInTheDocument();
    expect(screen.queryByRole("button")).not.toBeInTheDocument();
  });

  it("renders edit link and vote buttons for trip members", () => {
    renderWithAllProviders(
      <ActivityCard activity={testActivity} isTripMember={true} />
    );
    expect(screen.getByTitle("Edit Activity")).toBeInTheDocument();
    const buttons = screen.getAllByRole("button");
    expect(buttons.length).toBe(2);
  });
});
