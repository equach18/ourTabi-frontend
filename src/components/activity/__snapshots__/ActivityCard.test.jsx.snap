// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActivityCard component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="border border-zinc-300 p-4 rounded-lg shadow-sm hover:shadow-md transition bg-white"
  >
    <div
      class="flex justify-between items-start gap-4"
    >
      <div
        class="flex-1 space-y-1"
      >
        <div
          class="flex items-center gap-2"
        >
          <h3
            class="text-xl font-bold text-zinc-800"
          >
            Kayaking
          </h3>
          <a
            class="text-yellow-500 hover:text-yellow-600 transition"
            data-discover="true"
            href="/trips/53/activities/1/edit"
            title="Edit Activity"
          >
            <svg
              class="w-5 h-5"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m16.862 4.487 1.687-1.688a1.875 1.875 0 1 1 2.652 2.652L6.832 19.82a4.5 4.5 0 0 1-1.897 1.13l-2.685.8.8-2.685a4.5 4.5 0 0 1 1.13-1.897L16.863 4.487Zm0 0L19.5 7.125"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </a>
        </div>
        <p
          class="text-sm text-zinc-600"
        >
          <strong>
            Category:
          </strong>
           Other
        </p>
        <p
          class="text-sm text-zinc-600"
        >
          <strong>
            Location:
          </strong>
           Chicago
        </p>
        <p
          class="text-sm text-zinc-500"
        >
          <strong>
            Description:
          </strong>
           So much fun
        </p>
        <p
          class="text-sm text-zinc-500"
        >
          <strong>
            Scheduled:
          </strong>
           7/10/25, 9:30 AM
        </p>
      </div>
      <div
        class="flex items-center"
      >
        <div
          class="flex items-center space-x-2 mt-3"
        >
          <button
            class="px-2 py-2 rounded text-sm flex items-center gap-1 transition bg-gray-200 hover:bg-gray-300"
          >
            <svg
              class="size-5"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.633 10.25c.806 0 1.533-.446 2.031-1.08a9.041 9.041 0 0 1 2.861-2.4c.723-.384 1.35-.956 1.653-1.715a4.498 4.498 0 0 0 .322-1.672V2.75a.75.75 0 0 1 .75-.75 2.25 2.25 0 0 1 2.25 2.25c0 1.152-.26 2.243-.723 3.218-.266.558.107 1.282.725 1.282m0 0h3.126c1.026 0 1.945.694 2.054 1.715.045.422.068.85.068 1.285a11.95 11.95 0 0 1-2.649 7.521c-.388.482-.987.729-1.605.729H13.48c-.483 0-.964-.078-1.423-.23l-3.114-1.04a4.501 4.501 0 0 0-1.423-.23H5.904m10.598-9.75H14.25M5.904 18.5c.083.205.173.405.27.602.197.4-.078.898-.523.898h-.908c-.889 0-1.713-.518-1.972-1.368a12 12 0 0 1-.521-3.507c0-1.553.295-3.036.831-4.398C3.387 9.953 4.167 9.5 5 9.5h1.053c.472 0 .745.556.5.96a8.958 8.958 0 0 0-1.302 4.665c0 1.194.232 2.333.654 3.375Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            0
          </button>
          <button
            class="px-2 py-2 rounded text-sm flex items-center gap-1 transition bg-gray-200 hover:bg-gray-300"
          >
            <svg
              class="size-5"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.498 15.25H4.372c-1.026 0-1.945-.694-2.054-1.715a12.137 12.137 0 0 1-.068-1.285c0-2.848.992-5.464 2.649-7.521C5.287 4.247 5.886 4 6.504 4h4.016a4.5 4.5 0 0 1 1.423.23l3.114 1.04a4.5 4.5 0 0 0 1.423.23h1.294M7.498 15.25c.618 0 .991.724.725 1.282A7.471 7.471 0 0 0 7.5 19.75 2.25 2.25 0 0 0 9.75 22a.75.75 0 0 0 .75-.75v-.633c0-.573.11-1.14.322-1.672.304-.76.93-1.33 1.653-1.715a9.04 9.04 0 0 0 2.86-2.4c.498-.634 1.226-1.08 2.032-1.08h.384m-10.253 1.5H9.7m8.075-9.75c.01.05.027.1.05.148.593 1.2.925 2.55.925 3.977 0 1.487-.36 2.89-.999 4.125m.023-8.25c-.076-.365.183-.75.575-.75h.908c.889 0 1.713.518 1.972 1.368.339 1.11.521 2.287.521 3.507 0 1.553-.295 3.036-.831 4.398-.306.774-1.086 1.227-1.918 1.227h-1.053c-.472 0-.745-.556-.5-.96a8.95 8.95 0 0 0 .303-.54"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            0
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
